### ISSUE TITLE
v0.8.0 fails on both 0.18.0 and 0.19.0

### ISSUE THREAD

**Issue by genotrance** [nitely/nim-regex#23]
Simply running nimble test in nim-regex repo:

```
...
[OK] tfindandcaptureall
[OK] tstarts_with
[OK] tends_with
[OK] tliterals
[OK] tchar_class
[OK] tascii_set
[OK] treplace
[OK] tcompiletime
[OK] tmisc
[OK] tlook_around
[OK] tpretty_errors
Traceback (most recent call last)
tests.nim(1373)          tests
regex.nim(2505)          find
regex.nim(2485)          findImpl
regex.nim(2353)          setRegexMatch
regex.nim(2029)          clear
SIGSEGV: Illegal storage access. (Attempt to read from nil?)
Error: execution of an external program failed: '/home/gt/programming/nim-regex/tests/tests '
stack trace: (most recent call last)
home/gt/programming/nim-regex/regex.nimble(17) testTask
home/gt/programming/nimdevel/lib/system/nimscript.nim(237) exec
home/gt/programming/nimdevel/lib/system/nimscript.nim(237, 7) Error: unhandled exception: FAILED: nim c -r -d:forceRegexAtRuntime tests/tests.nim
```

**Comment by genotrance** [comment]
Never mind, was using wrong nim build. However, nimgen tests are failing so there's something I'm doing with nim-regex that crashes but I don't have a simpler test case yet.

**Comment by nitely** [comment]
mmh I think I indeed broke it. I misconfigured travis and it was only testing Nim devel. I'll release a fix, but I think I'll just drop Nim 0.18 support in v0.9.

**Comment by nitely** [comment]
done
